juice/src/net/layer.rs

```diff
@@ -8,6 +8,12 @@ use crate::net::loss::*;
 use crate::net::{Context, Descriptor, LayerConfig};
 use crate::util::LayerOps;
 
+#[derive(Debug)]
+pub enum LayerError {
+    Backend(::coaster::error::Error),
```

Comment:
We should be careful not to return too large error types, after all this is the hot code path and we don't expect these to happen often. If we start nesting too many, this might increase stack usage - by the fact that the `Result` type gets blown up - significantly.

I generally recommend `thiserror` to avoid boilerplate and provide `std::error::Error` implementations as well as `std::fmt::Display`.

Reply:
So if possible, wrapping the variant inner types in a `Box` would make the type smaller while keeping the matchable enum semantics.

Reply:
Something like this?

Reply:
Resolving.